code/modules/power/generator.dm

```diff
@@ -69,8 +69,8 @@
 				var/heat = energy_transfer*(1-efficiency)
 				lastgen += energy_transfer*efficiency
 
-				hot_air.temperature = hot_air.temperature - energy_transfer/hot_air_heat_capacity
-				cold_air.temperature = cold_air.temperature + heat/cold_air_heat_capacity
+				hot_air.set_temperature(hot_air.return_temperature() - energy_transfer/hot_air_heat_capacity)
+				cold_air.set_temperature(cold_air.return_temperature() + heat/cold_air_heat_capacity)
```

Comment:
Shouldn't this be a usecase for adjust temp? Or have I forgotten PEMDAS/Does such a thing not exist